runtime/common/src/asset_registry.rs

```diff
@@ -54,3 +56,24 @@ impl AssetProcessor<CurrencyId, AssetMetadata<Balance, CustomMetadata>> for Cust
 }
 
 pub type AssetAuthority = AsEnsureOriginWithArg<EnsureRoot<AccountId>>;
+pub struct FixedConversionRateProvider<OrmlAssetRegistry, CurrencyIdConvert>(PhantomData<(OrmlAssetRegistry,CurrencyIdConvert)>);
+
+impl<
+		OrmlAssetRegistry: Inspect<
+			AssetId = CurrencyId,
+			Balance = Balance,
+			CustomMetadata = asset_registry::CustomMetadata,
+		>,
+		CurrencyIdConvert: Convert<MultiLocation, Option<CurrencyId>>,
+	> FixedConversionRateProviderTrait for FixedConversionRateProvider<OrmlAssetRegistry, CurrencyIdConvert>
+{
+	fn get_fee_per_second(location: &MultiLocation) -> Option<u128> {
+		let asset_id_maybe = CurrencyIdConvert::convert(location.clone());
```

Comment:
Instead of looking the asset registry by `MultiLocation` we look the metadata by `CurrencyId`. Although this is the same in most cases, it removes the need to handle different `MultiLocation` representations for the same asset, which is a problem since we can only store one on the asset registry.
We therefore will delegate this to the `CurrencyIdConvert` which will normalize it as unanchored.

Example: we need to handle Native, which is both `{parent: 0, interior: PalletId(10)}`, `{parent: 1, interior: ParachainId(id), PalletId(10)}`

Reply:
UPDATE: this is now outdated and there is no need to perform this conversion.